stdio/file.c

```diff
@@ -845,19 +846,14 @@ ssize_t getline(char **lineptr, size_t *n, FILE *stream)
 	if (!nl)
 		linesz++;
 
-	if (*lineptr == NULL)
-		*lineptr = malloc(linesz);
-	else if (*n < linesz) {
-		if ((tmp = realloc(*lineptr, linesz)) == NULL) {
-			free(*lineptr);
-		}
+	if (*lineptr == NULL || *n < linesz) {
+		tmp = (*lineptr ? realloc(*lineptr, linesz) : malloc(linesz));
```

Comment:
`realloc(NULL, linesz)` is the equivalent of `malloc(linesz)`, so the line may be simplified:
```suggestion
		tmp = realloc(*lineptr, linesz);
```

Reply:
Changed to sole `realloc()`. Thanks!